api/wasm/cpp/proxy_wasm_externs.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Intrinsic functions for WASM modules.
+ */
+// NOLINT(namespace-envoy)
+
+#pragma once
+
+#include "stddef.h"
+
+//
+//  ABI calls into the VM.
+//
+
+// Non-stream calls.
+extern "C" EMSCRIPTEN_KEEPALIVE uint32_t proxy_on_start(uint32_t root_context_id,
```

Comment:
Why aren't we following Envoy style conventions in our SDK APIs?

Reply:
This is the ABI and it is following the WASI style since that is the standardization body which we are working with for the ABI.

Reply:
As per @mattklein123 comment, this is the first time this has come up, it would be good to understand more details. Given the key role this ABI is going to play in Envoy going forward beyond just WASM, it would be good to align on strategy for how this ABI is going to be standardized, governed and evolved.